bin/node/runtime/src/lib.rs

```diff
@@ -468,11 +468,102 @@ parameter_types! {
 	pub const SlashDeferDuration: pallet_staking::EraIndex = 24 * 7; // 1/4 the bonding duration.
 	pub const RewardCurve: &'static PiecewiseLinear<'static> = &REWARD_CURVE;
 	pub const MaxNominatorRewardedPerValidator: u32 = 256;
-}
+	// enable the below flag to allow dynamic damping of validator count per era.
+	pub const EnableAutomaticValidatorUpdatePerEra: bool = true;
+	pub const BottomXPercentOfValidators: u32 = 20;
+	pub const BottomYPercentOfValidators: u32 = 40;
+}
+
+pub struct AutomaticValidatorUpdatePerEra;
+impl Convert<(u32, Vec<BalanceOf<T>>, u32, u32, u32, u32), u32> for AutomaticValidatorUpdatePerEra {
```

Comment:
```suggestion
impl Convert<(u32, Vec<Balance>, u32, u32, u32, u32), u32> for AutomaticValidatorUpdatePerEra {
```

Exercise: You should trace back how `type BalanceOf<T> = ...` traces back to `type Balance = Balance` in this file in `balances::Config`, and also realize that this is where we resolve it to a concrete type, u128.